crates/sui-adapter/src/execution_engine.rs

```diff
@@ -376,6 +385,53 @@ pub fn construct_advance_epoch_pt(
     Ok(builder.finish())
 }
 
+pub fn construct_advance_epoch_safe_mode_pt(
```

Comment:
GitHub doesn't let me comment on code that did not change but the comment on line 377 should be changed to say "Step 3"